### PR TITLE
BSI docs

### LINKED ISSUE #740: BSI Range-Encoding: add documentation
- All new PQL functionality should be described in the query language section.
- A brief overview of range encoding storage and operations should be added to the data model section.
- A brief usage example should be added to the tutorial section (taxi data could work for this).

### LINKED ISSUE #853: Add documentation for Xor call.
### Description
`Xor` is a new function call, and we have yet to provide documentation and examples.

### Success criteria (What criteria will consider this ticket closeable?)
`Xor` is represented in the docs.

### PR DESCRIPTION
**Pull request by raskle**
## Overview
BSI range encoding docs.
This work is dependent on Range Encoding additions to the Getting Started project

Fixes #740 
Fixes #853 

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: docs/api-reference.md [hunk 1]
```diff
@@ -162,6 +162,14 @@ The request payload is in JSON, and may contain the `options` field. The `option
 * `inverseEnabled` (boolean): Enables [the inverted view]({{< ref "data-model.md#inverse" >}}) for this frame if `true`.
 * `cacheType` (string): [ranked]({{< ref "data-model.md#ranked" >}}) or [LRU]({{< ref "data-model.md#lru" >}}) caching on this frame. Default is `lru`.
 * `cacheSize` (int): Number of rows to keep in the cache. Default 50,000.
+* `rangeEnabled` (boolean): Enables range encoded numbers in this frame.
```
Comment (alanbernstein) [inline docs/api-reference.md]: Would "fields" or something be better than "numbers" here?

File: docs/api-reference.md [hunk 2]
```diff
@@ -162,6 +162,14 @@ The request payload is in JSON, and may contain the `options` field. The `option
...
+* `fields` (array): List of range encoded fields.
+
+Each individual `field` contains the following:
+* `name` (string): Field name.
+* `type` (string): Field type, currently only "int" is supported.
+* `min` (int): Minimum of the value range stored in this field.
+* `max` (int): Maximum of the value range stored in this field.
```
Comment (alanbernstein) [inline docs/api-reference.md]: What is the range of allowed values of max and min? In particular, are negative values allowed?

Comment (alanbernstein) [reply docs/api-reference.md]: We might point out that a field will use `ceil(log2(max-min))` bits for the value, so the user could maximize the allowed values by using a power of two range.

Comment (alanbernstein) [reply docs/api-reference.md]: Which of these can be modified, if any?

Comment (benbjohnson) [reply docs/api-reference.md]: Yes, negative values are allowed. Min and max can be anything between `MinInt64` and `MaxInt64`. These values currently cannot be modified. Changing `min` would require re-encoding everything. Increasing `max` could be possible. `name` could be changed too but it requires renaming all the views.

**Comment by codysoyland** [comment]
Closing in favor of #861

**Pull request closed without merging**